packages/ui/src/Checkbox.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ChangeEvent, FC, FocusEvent, useEffect, useRef } from 'react'
+import styles from './Checkbox.module.scss'
+import { Check, Minus } from 'react-feather'
+import { DataTestProp } from '@hazelcast/helpers'
+import classNames from 'classnames'
+import { Error, errorId } from './Error'
+import { Help, helpTooltipId } from './Help'
+import { v4 as uuid } from 'uuid'
+
+type CheckboxCoreProps = {
+  name: string
+  value?: string
+  onBlur?: (e: FocusEvent<HTMLInputElement>) => void
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void
+  error?: string
+  checked?: boolean
+}
+
+export type CheckboxExtraProps = {
+  label: string
+  helperText?: string
+  indeterminate?: boolean
+  disabled?: boolean
+  required?: boolean
+  className?: string
+}
+
+type CheckboxProps = CheckboxCoreProps & CheckboxExtraProps & DataTestProp
+
+/**
+ * ### Purpose
+ * Forms require input from users. If you need information that can be represented as boolean, use input with type 'checkbox'.
+ * Checkbox fields accept boolean types of data. And help and error guidance to ensure they know what to enter.
+ *
+ *
+ * ### General Info
+ * - It's important to realise that we don't set checkbox value, but state (on/off). That is the main difference from other input types.
+ * - They can have a special indeterminate state, that represents a '3rd' value, usually used in tree structures, etc.
+ */
+export const Checkbox: FC<CheckboxProps> = ({
+  checked,
+  name,
+  onChange,
+  onBlur,
+  className,
+  value,
+  indeterminate = false,
+  label,
+  helperText,
+  error,
+  disabled = false,
+  required,
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null)
+  const idRef = useRef(uuid())
+
+  // we want to support indeterminate as a React property, but it needs to be set programatically
+  useEffect(() => {
+    if (inputRef?.current) {
+      inputRef.current.indeterminate = indeterminate
+    }
+  })
+
+  return (
+    <span>
```

Comment:
@aigoncharov I'd vote for span, what if you want to display checkboxes as options next to each other? You can still wrap them in block based component, but it's not that easy the otherway around.

Reply:
@jvorcak `Error` inside of the `span` is displayed as a `block`. Making the parent `inline` does not have any effect because of it. The browser now has to guess and deduce the correct display value.

Reply:
@aigoncharov regular checkbox we try to emulate is also not block-based

Reply:
If you want to display checkboxes next to each other, you just make the parent container `flex`. Wouldn't it work?

Reply:
@jvorcak our checkboxes already do not behave `inline`.
Consider this story
```
export const Default = () => {
  const [checked, setChecked] = useState<boolean>(true)
  return (
    <p>
      <Checkbox
        name="default"
        checked={checked}
        label="Label"
        onChange={(e) => {
          setChecked(e.target.checked)
        }}
        error="11111"
      />
      <Checkbox
        name="default"
        checked={checked}
        label="Label"
        onChange={(e) => {
          setChecked(e.target.checked)
        }}
        error="222"
      />
    </p>
  )
}
```
See how it is displayed in Chrome.
Now remove one of the errors. But only one. Now remove the second one. 
If we keep the checkbox container as an `inline` element, browser has to forcefully make it `block` when error messages appear. The problem is not even with that. The problem is that now the behavior of our checkbox is dynamic. If it does not have an error, then it is inline. If it does, it is block.

Reply:
Fair point, updated.

Reply:
@jvorcak AFAICS, it is still a `span`. Have you pushed yet?

Reply:
@aigoncharov I've. I can see it's a div on github https://github.com/hazelcast/frontend-shared/pull/38/files#diff-f74d41384ec798601c50a242cf943dd1fb7d727683664c7ce7e5ee5c6ac766d5R67

Reply:
Sorry. For some reason I did not see that commit at the time of my comment